src/deadline_worker_agent/worker.py

```diff
@@ -246,6 +246,10 @@ def run(self) -> None:
                                 grace_time=worker_shutdown.grace_time,
                                 fail_message=worker_shutdown.fail_message,
                             )
+                        else:
+                            # If we are here, it's because self._stop.set() was set causing the monitor_ec2_shutdown thread to join.
+                            # The scheduler thread has a longer wait, so let's wake it up so it can join as well.
+                            self._scheduler._wakeup.set()
```

Comment:
Should we be using:

```
self._scheduler.shutdown(...)
```

instead? It is marked as a public API and allows providing a reason to the tasks being interrupted.

Reply:
Updated and re-tested!